src/lib/mina_base/user_command.ml

```diff
@@ -404,24 +404,31 @@ end
 
 let check_well_formedness ~genesis_constants t :
     (unit, Well_formedness_error.t list) result =
-  let preds =
-    let open Well_formedness_error in
-    [ (has_insufficient_fee, Insufficient_fee)
-    ; (has_zero_vesting_period, Zero_vesting_period)
-    ]
-  in
-  let errs0 =
-    List.fold preds ~init:[] ~f:(fun acc (f, err) ->
-        if f t then err :: acc else acc )
-  in
-  let errs =
-    match valid_size ~genesis_constants t with
-    | Ok () ->
-        errs0
-    | Error err ->
-        Zkapp_too_big err :: errs0
-  in
-  if List.is_empty errs then Ok () else Error errs
+  if Mina_compile_config.zkapps_disabled then
```

Comment:
This logic is not correct, we still need to check for the fee. Add this check as another predicate in `preds` instead.

Reply:
Thanks for the quick feedback @mrmr1993.